tbdex/rfq_create.go

```diff
@@ -0,0 +1,226 @@
+	}
+
+
+    h := rfqHashes{
+        PayinHash: payinHash,
+        PayoutHash: payoutHash,
+        ClaimsHash: claimsHash,
+    }
+
+    p := RFQPrivateData {
+        Salt: salt,
+        Payin: &PrivatePaymentDetails{
+            PaymentDetails: payin.PaymentDetails,
+        },
+        Payout: &PrivatePaymentDetails{
+            PaymentDetails: payout.PaymentDetails,
+        },
+        Claims: claims,
+    }
+
+    return h, p, nil
+}
+
+func digestData(salt string, data any) (string, error) {
+    digestible := []interface{}{salt, data}
+
+	byteArray, err := Digest(digestible)
+	if err != nil {
+		return "", err
+	}
+
+	encodedString := base64.URLEncoding.EncodeToString(byteArray)
+
+	return encodedString, nil
+}
+
+// PayinMethodWithPrivate is used to create the payin method for an RFQ
+type PayinMethodWithPrivate struct {
+	Amount         string         `json:"amount"`
+	Kind           string         `json:"kind"`
+	PaymentDetails map[string]any `json:"paymentDetails"`
+}
+
+// PayoutMethodWithPrivate is used to create the payout method for an RFQ
+type PayoutMethodWithPrivate struct {
+	Kind           string         `json:"kind"`
+	PaymentDetails map[string]any `json:"paymentDetails"`
+}
+
+type rfqHashes struct {
+	PayinHash  string
+	PayoutHash string
+	ClaimsHash string
+}
+
+type createRFQOptions struct {
+	id         string
+	createdAt  time.Time
+	protocol   string
+	externalID string
+	claims     []string
+}
+
+// CreateRFQOption is a function type used to apply options to RFQ creation.
+type CreateRFQOption func(*createRFQOptions)
+
+// WithRFQID can be passed to [CreateRFQ] to provide a custom id.
+func WithRFQID(id string) CreateRFQOption {
+	return func(r *createRFQOptions) {
+		r.id = id
+	}
+}
+
+// WithRFQCreatedAt can be passed to [CreateRFQ] to provide a custom created at time.
+func WithRFQCreatedAt(t time.Time) CreateRFQOption {
+	return func(r *createRFQOptions) {
+		r.createdAt = t
+	}
+}
+
+// WithRFQExternalID can be passed to [CreateRFQ] to provide a custom external id.
+func WithRFQExternalID(externalID string) CreateRFQOption {
+	return func(r *createRFQOptions) {
+		r.externalID = externalID
+	}
+}
+
+// WithRFQClaims can be passed to [CreateRFQ] to provide a custom external id.
+func WithRFQClaims(claims []string) CreateRFQOption {
+	return func(r *createRFQOptions) {
+		r.claims = claims
+	}
+}
+
+// PayinMethodWithPrivateOption is a function type used to apply options to [PayinMethodWithPrivate] creation.
+type PayinMethodWithPrivateOption func(*PayinMethodWithPrivate)
+
+// PayoutMethodWithPrivateOption is a function type used to apply options to [PayoutMethodWithPrivate] creation.
+type PayoutMethodWithPrivateOption func(*PayoutMethodWithPrivate)
+
+// WithPayinMethodWithPrivate can be passed to [WithRFQSelectedPayinMethod] to provide a hash of the payment details.
+func WithPayinMethodWithPrivate(detailsPrivate map[string]any) PayinMethodWithPrivateOption {
```

Comment:
i don't think we have to surface `Private` concept to callers.  we can tuck the entire thing away